workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/FieldTypes/OptionFieldType.cs

```diff
@@ -86,15 +86,12 @@ public string GetDeserializeUpdateIntoUpdateString(string updateFieldInstance, s
 
         public string GetDeserializeDataIntoUpdateString(string updateFieldInstance, string schemaObject, uint fieldNumber)
         {
-            return new CustomScopeBlock(scope =>
-            {
-                scope.If($"{containedType.GetCountExpression(schemaObject, fieldNumber)} == 1",
-                    then =>
-                    {
-                        then.Line(
-                            $"{updateFieldInstance} = new global::Improbable.Gdk.Core.Option<{Type}>({containedType.GetDeserializationExpression(schemaObject, fieldNumber)});");
-                    });
-            }).Format();
+            return new IfElseBlock($"{containedType.GetCountExpression(schemaObject, fieldNumber)} == 1",
```

Comment:
Its odd that the `IfElseBlock` doesn't always have an else

Reply:
generators won't have any idea of what blocks there are - they'll only be able to create blocks through the methods such as `.If`/`.Line` etc

Constructors like these are internal only - so if a user wants to create individual blocks (useful when splitting up one large generator into several methods), there'll be a static `Scope` class with helpers similar to `.If`/`.Method`/`.Type` etc for creating the blocks e.g. `Scope.If`